dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -170,6 +170,12 @@ def authenticate
       destination_url = "#{target_link_uri}?#{redirect_params.to_query}"
 
       if user
+        # If this is the user's first login, send them into the account linking flow
+        if user.sign_in_count < 1 && DCDO.get('lti_account_linking_enabled', false)
+          PartialRegistration.persist_attributes(session, user)
```

Comment:
I also tried caching the user directly using `CDO.shared_cache` instead of using `PartialRegistration`. Since it was essentially doing the exact same thing but under a different cache key, it felt cleaner to just use a `PartialRegistration`. Plus this works with the rest of the code that checks for in-progress partial registrations. It also doesn't feel that far off semantically, since roster sync users who haven't logged in technically haven't finished the registration process; they still need to go through the account linking flow.

If anyone has strong preferences here, or better ideas, let me know.

Reply:
You'll want to test this with the student email remediation DCDO as well since email will be removed in that mode

Reply:
Good call, will test that out